haddock-api/src/Haddock/Types.hs

```diff
@@ -388,7 +388,7 @@ mkPseudoFamilyDecl (FamilyDecl { .. }) = PseudoFamilyDecl
 
 
 -- | An instance head that may have documentation and a source location.
-type DocInstance name = (InstHead name, Maybe (MDoc (IdP name)), Located (IdP name))
+type DocInstance name = (InstHead name, Maybe (MDoc (IdP name)), Located (IdP name), Maybe Module)
```

Comment:
Unfortunately this is a breaking change :( This has to go into the next major version.

Reply:
I suspected as much... Should I reopen this PR on `ghc-head` then?